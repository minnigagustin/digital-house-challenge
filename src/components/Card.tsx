import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { scale, verticalScale } from "utils/index";

interface Props {
  points: number;
  month: string;
}

export function Card({ month, points }: Props) {
  return (
    <View style={styles.container}>
      <Text style={styles.date}>{month}</Text>
      <Text style={styles.points}>{points} pts</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    paddingTop: verticalScale(8),
    backgroundColor: "#334FFA",
    height: verticalScale(120),
    width: "80%",
    borderRadius: scale(20),
    paddingHorizontal: scale(19),
    shadowColor: "black",
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,
    elevation: 5,
  },
  date: {
    position: "absolute",
    left: scale(19),
    top: verticalScale(21),
    color: "white",
    fontSize: 16,
    fontWeight: "800",
    textTransform: "capitalize",
    fontFamily: "AvenirBold",
  },
  points: {
    color: "white",
    fontWeight: "bold",
    fontSize: scale(32),
    fontFamily: "AvenirBold",
  },
});
