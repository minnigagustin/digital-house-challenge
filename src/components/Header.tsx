import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { scale, verticalScale } from "utils/index";

interface Props {
  name: string;
}

export function Header({ name }: Props) {
  return (
    <View>
      <Text style={styles.welcome}>Bienvenido de vuelta!</Text>
      <Text style={styles.userName}>{name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  welcome: {
    fontSize: scale(16),
    fontWeight: "800",
    fontFamily: "AvenirBold",
  },
  userName: {
    fontSize: scale(14),
    fontFamily: "AvenirRegular",
    lineHeight: verticalScale(22),
  },
});
