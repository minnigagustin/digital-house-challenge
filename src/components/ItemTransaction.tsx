import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { Image } from "expo-image";
import React from "react";
import { Text, View, StyleSheet, Pressable } from "react-native";
import { formatDate } from "utils/index";

export interface Props {
  date: Date;
  product: string;
  points: number;
  image: string;
  is_redemption: boolean;
  id: string;
}

export function ItemTransaction({
  image,
  product,
  date,
  is_redemption,
  points,
}: Props) {
  const { navigate } = useNavigation<NativeStackNavigationProp<any>>();

  const handlePress = () =>
    navigate("ProductDetails", {
      date,
      image,
      points,
      product,
    });

  return (
    <Pressable onPress={handlePress}>
      <View style={styles.container}>
        <>
          <Image
            style={styles.image}
            source={{ uri: image }}
            priority={"high"}
          />
          <View style={styles.infoContainer}>
            <Text style={styles.product}>{product}</Text>
            <Text style={styles.date}>{formatDate(date)}</Text>
          </View>
        </>
        <>
          {is_redemption ? (
            <Text style={styles.min}>
              -<Text style={styles.points}>{`${points}  >`}</Text>
            </Text>
          ) : (
            <Text style={styles.sum}>
              +<Text style={styles.points}>{`${points}  >`}</Text>
            </Text>
          )}
        </>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    padding: 10,
  },
  image: {
    width: 55,
    height: 55,
    marginRight: 11,
    borderRadius: 10,
  },
  infoContainer: {
    flex: 1,
  },
  product: {
    fontSize: 14,
    fontWeight: "800",
    color: "#000",
    fontFamily: "AvenirBold",
  },
  date: {
    fontSize: 12,
    fontWeight: "400",
    marginTop: 7,
    color: "#000",
    fontFamily: "AvenirMedium",
  },
  min: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#FF0000",
  },
  sum: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#00B833",
  },
  points: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#000",
  },
});
