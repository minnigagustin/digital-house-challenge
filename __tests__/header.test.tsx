import { Header } from "components/Header";
import React from "react";
import renderer from "react-test-renderer";
const name = "Ruben Rodriguez";
describe("Header", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Header name={name} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
