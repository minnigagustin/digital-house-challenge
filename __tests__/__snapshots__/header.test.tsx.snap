// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<View>
  <Text
    style={
      {
        "fontFamily": "AvenirBold",
        "fontSize": 34.285714285714285,
        "fontWeight": "800",
      }
    }
  >
    Bienvenido de vuelta!
  </Text>
  <Text
    style={
      {
        "fontFamily": "AvenirRegular",
        "fontSize": 30,
        "lineHeight": 43.15882352941176,
      }
    }
  >
    Ruben Rodriguez
  </Text>
</View>
`;
